python.wake

```diff
@@ -24,13 +24,16 @@
 
 ######################################################
 # Add a Python virtual environment to a plan.
-#   Currently only supports pipenv, but this is where would examine the directory
-#   to see if other env managers are being used.
 ##########################################################
 global def addPythonEnv pipDir plan =
+  def requirements = `{pipDir}/requirements.txt`
 
   # Install the virtual environment if not already done
-  def installed = installPipenvEnv pipDir
+  def installed = match (sources pipDir requirements)
+    _, Nil = installPythonRequirementsEnv pipDir
+    Nil = installPipenvEnv pipDir
+    _ = panic "unreachable: multiple sources match requirements.txt"
```

Comment:
Does this make sense as a way to make `addPythonEnv` requirements/pipenv agnostic?

Reply:
I would say that it's probably not worth the effort to make `addPythonEnv` work for both pipenv and `requirements.txt` flows. I think it's perfectly fine to have two separate functions `addPythonEnv` and `addPythonRequirementsEnv` functions.